File: post/leetcode/cpp/remove-duplicates-from-sorted-array.cpp
/*
@Copyright:LintCode
@Author:   justice_103
@Problem:  http://www.lintcode.com/problem/remove-duplicates-from-sorted-array
@Language: C++
@Datetime: 15-07-22 12:45
*/

class Solution {
public:
    /**
     * @param A: a list of integers
     * @return : return an integer
     */
    int removeDuplicates(vector<int> &nums) {
        // write your code here
        //:

        const int n = nums.size();
        if(n<2) return n;

        int t=0;
        for(int i=1; i<n; ++i){
            if(nums[i]!=nums[t])
                nums[++t]=nums[i];
        }
        return t+1;
    }
};

File: post/leetcode/python/remove-duplicates-from-sorted-array.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   justice_103
@Problem:  http://www.lintcode.com/problem/remove-duplicates-from-sorted-array
@Language: Python
@Datetime: 15-08-23 03:47
'''

class Solution:
    """
    @param A: a list of integers
    @return an integer
    """
    def removeDuplicates(self, A):
        # write your code here
        n=len(A)
        
        if n<2:
            return n
        
        j=0
        for i in range(1,n):
            if A[i]!=A[j]:
                j+=1
                A[j]=A[i]
            
        return j+1
        
        
